core/server/adapters/scheduling/post-scheduling/index.js

```diff
@@ -1,24 +1,79 @@
-const Promise = require('bluebird'),
-    moment = require('moment'),
-    localUtils = require('../utils'),
-    common = require('../../../lib/common'),
-    models = require('../../../models'),
-    urlUtils = require('../../../lib/url-utils'),
-    _private = {};
+const Promise = require('bluebird');
+const moment = require('moment');
+const jwt = require('jsonwebtoken');
+const localUtils = require('../utils');
+const common = require('../../../lib/common');
+const models = require('../../../models');
+const urlUtils = require('../../../lib/url-utils');
+const _private = {};
+const SCHEDULED_RESOURCES = ['post', 'page'];
+
+/**
+ * @description Load the internal scheduler integration
+ *
+ * @return {Promise}
+ */
+_private.getSchedulerIntegration = function () {
+    return models.Integration.findOne({slug: 'ghost-scheduler'}, {withRelated: 'api_keys'})
+        .then((integration) => {
+            if (!integration) {
+                throw new common.errors.NotFoundError({
+                    message: common.i18n.t('errors.api.resource.resourceNotFound', {
+                        resource: 'Integration'
+                    })
+                });
+            }
+            return integration.toJSON();
+        });
+};
+
+/**
+ * @description Get signed admin token for making authenticated scheduling requests
+ *
+ * @return {Promise}
+ */
+_private.getSignedAdminToken = function (options) {
```

Comment:
This whole method is a pretty good candidate to get out into an SDK

Reply:
Maybe a future refactor, don't think it's necessary for this PR 🙂